### PR TITLE
feat(site-builder): improve site builder output prints and errors

### LINKED ISSUE #65: Provide more INFO output and better errors in site-builder
Currently, the site builder is very quiet. After telling me which wallet is used, it stops printing output. And, depending on the number and size of files, it remains quiet for a long time.

It would be helpful to provide info output like "storing file X" to show the user what is happening. Or at least provide some other progress output.

Also, the errors could be improved. For example for the case where the `walrus` binary cannot be found.

### PR DESCRIPTION
**Pull request by giac-mysten**
A few UI/UX improvements:

- Progress is marked clearly, especially the uploads to Walrus;
- The source of errors in the execution is better identified, especially in the case of
  - empty files;
  - unsupported file extension;
  - missing/inaccessible walrus binary.

Related to #88 
Closes #65 
Contributes to #92

### CODE REVIEW

File: site-builder/src/display.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            PrintStyledContent(format!("\n{message}\n").green().bold()),
+        )
+        .unwrap();
+    }
+}
+
+pub fn error<S: Display>(message: S) {
+    if cfg!(not(test)) {
+        crossterm::execute!(
+            stdout(),
```
Comment (mlegner) [inline site-builder/src/display.rs:25]: Suggestion: Use `stderr` here.

File: site-builder/src/site/content.rs [hunk 2]
```diff
@@ -121,89 +121,92 @@ pub enum ContentType {
...
+            "woff" => Ok(ContentType::FontWoff),
+            "woff2" => Ok(ContentType::FontWoff2),
+            "xhtml" => Ok(ContentType::ApplicationXhtmlxml),
+            "xls" => Ok(ContentType::ApplicationVndmsexcel),
+            "xlsx" => Ok(ContentType::ApplicationVndopenxmlformatsofficedocumentspreadsheetmlsheet),
+            "xml" => Ok(ContentType::ApplicationXml),
+            "xul" => Ok(ContentType::ApplicationVndmozillaxulxml),
+            "zip" => Ok(ContentType::ApplicationZip),
+            "7z" => Ok(ContentType::ApplicationX7zcompressed),
+            _ => Err(anyhow!(
```
Comment (mlegner) [inline site-builder/src/site/content.rs:206]: Hint: If you change this to `return Err(...)`, you can remove the `Ok()` from all the other cases and just wrap the whole thing into an `Ok`.

File: site-builder/src/site/resource.rs [hunk 3]
```diff
@@ -330,13 +333,17 @@ impl ResourceManager {
...
     ) -> Result<Option<Resource>> {
-        let content_type = ContentType::from_extension(
+        let extension = full_path.extension().unwrap_or(
             full_path
-                .extension()
-                .ok_or(anyhow!("No extension found for {:?}", full_path))?
-                .to_str()
-                .ok_or(anyhow!("Invalid extension"))?,
+                .file_name()
+                .expect("the path should no terminate in `..`"),
```
Comment (mlegner) [inline site-builder/src/site/resource.rs:339]: Typo:
```suggestion
                .expect("the path should not terminate in `..`"),
```

File: site-builder/src/site/resource.rs
Comment (mlegner) [inline site-builder/src/site/resource.rs:1]: Remark: I don't see any changes to empty files. Maybe we can at least print a message that we are ignoring them?

**Review by mlegner — APPROVED** [review]
Thanks for these UX improvements, @giac-mysten. Just a few minor comments.

File: site-builder/src/site/resource.rs
Comment (giac-mysten) [reply site-builder/src/site/resource.rs:1]: I had kept the error for empty files (see image above). My thinking was that it's better to error out and let the user consider that there is an empty file that cannot be uploaded, and it's up to them to decide what to do.

I can change this to a warning though.

Comment (mlegner) [reply site-builder/src/site/resource.rs:1]: Ah, sorry, got confused about this. This works until we have a better support for empty files. 👍

File: site-builder/src/display.rs (hunk 1 above)
Comment (mlegner) [reply site-builder/src/display.rs:25]: I don't see a change here, did you forget to push something?

Comment (giac-mysten) [reply site-builder/src/display.rs:25]: I haven't yet pushed  :)

Comment (mlegner) [reply site-builder/src/display.rs:25]: Ah, sorry, I was a bit hasty... 😅

**Pull request merged**